### ISSUE TITLE
Use of uninitialized value $batch_start in subtraction trimmomatic_gzip error

### ISSUE THREAD

**Issue by nizardrou** [jerowe/HPC-Runner-Command#12]
I get this error with updating task dependencies. The error is generated at the trimmomatic gzip command submissions step. Below is the code,
```
# Starting trimmomatic_gzip
#



#
# Variables 
# Indir: /scratch/gencore/170518_SN7001341_0237_ACA6CHACXX/Unaligned/Project_PAG-8/{$self->trimmomatic}
# Outdir: /scratch/gencore/170518_SN7001341_0237_ACA6CHACXX/Unaligned/Project_PAG-8/data/processed/trimmomatic_gzip
# Local Variables:
#	indir: {$self->trimmomatic}
#	INPUT: {$self->indir}/{$sample}
#	before_meta: HPC Directives

#
#HPC jobname=trimmomatic_gzip
#HPC deps=trimmomatic_fastqc
#HPC account=gencore
#HPC mem=60GB
#HPC module=gencore/1 gencore_dev gencore_qc
#HPC walltime=25:15:00
#HPC cpus_per_task=1
#HPC commands_per_node=2
#HPC procs=1
#HPC partition=serial
#

#	outdir: /scratch/gencore/170518_SN7001341_0237_ACA6CHACXX/Unaligned/Project_PAG-8/data/processed/trimmomatic_gzip
#	OUTPUT: {$self->outdir}/{$sample}
#

#TASK tags=Sample_PAG008_V4_E2
gzip -f /scratch/gencore/170518_SN7001341_0237_ACA6CHACXX/Unaligned/Project_PAG-8/data/processed/Sample_PAG008_V4_E2/trimmomatic/Sample_PAG008_V4_E2_read1_trimmomatic_1PE.fastq

#TASK tags=Sample_PAG008_V4_E2
gzip -f /scratch/gencore/170518_SN7001341_0237_ACA6CHACXX/Unaligned/Project_PAG-8/data/processed/Sample_PAG008_V4_E2/trimmomatic/Sample_PAG008_V4_E2_read2_trimmomatic_2PE.fastq

#TASK tags=Sample_PAG008_V4_E2
gzip -f /scratch/gencore/170518_SN7001341_0237_ACA6CHACXX/Unaligned/Project_PAG-8/data/processed/Sample_PAG008_V4_E2/trimmomatic/Sample_PAG008_V4_E2_read1_trimmomatic_1SE.fastq

#TASK tags=Sample_PAG008_V4_E2
gzip -f /scratch/gencore/170518_SN7001341_0237_ACA6CHACXX/Unaligned/Project_PAG-8/data/processed/Sample_PAG008_V4_E2/trimmomatic/Sample_PAG008_V4_E2_read2_trimmomatic_2SE.fastq


#TASK tags=Sample_PAG025_V3_E2
gzip -f /scratch/gencore/170518_SN7001341_0237_ACA6CHACXX/Unaligned/Project_PAG-8/data/processed/Sample_PAG025_V3_E2/trimmomatic/Sample_PAG025_V3_E2_read1_trimmomatic_1PE.fastq

#TASK tags=Sample_PAG025_V3_E2
gzip -f /scratch/gencore/170518_SN7001341_0237_ACA6CHACXX/Unaligned/Project_PAG-8/data/processed/Sample_PAG025_V3_E2/trimmomatic/Sample_PAG025_V3_E2_read2_trimmomatic_2PE.fastq

#TASK tags=Sample_PAG025_V3_E2
gzip -f /scratch/gencore/170518_SN7001341_0237_ACA6CHACXX/Unaligned/Project_PAG-8/data/processed/Sample_PAG025_V3_E2/trimmomatic/Sample_PAG025_V3_E2_read1_trimmomatic_1SE.fastq

#TASK tags=Sample_PAG025_V3_E2
gzip -f /scratch/gencore/170518_SN7001341_0237_ACA6CHACXX/Unaligned/Project_PAG-8/data/processed/Sample_PAG025_V3_E2/trimmomatic/Sample_PAG025_V3_E2_read2_trimmomatic_2SE.fastq



#
# Ending trimmomatic_gzip
#
```
Specifically, the error is,

```
Use of uninitialized value $batch_start in subtraction (-) at /scratch/gencore/.local/easybuild/software/gencore_dev/1.0/lib/perl5/site_perl/5.22.0/HPC/Runner/Command/submit_jobs/Utils/Scheduler.pm line 1411.
Use of uninitialized value $batch_end in subtraction (-) at /scratch/gencore/.local/easybuild/software/gencore_dev/1.0/lib/perl5/site_perl/5.22.0/HPC/Runner/Command/submit_jobs/Utils/Scheduler.pm line 1411.
Use of uninitialized value $batch_start in concatenation (.) or string at /scratch/gencore/.local/easybuild/software/gencore_dev/1.0/lib/perl5/site_perl/5.22.0/HPC/Runner/Command/submit_jobs/Utils/Scheduler.pm line 1415.
Use of uninitialized value $batch_end in concatenation (.) or string at /scratch/gencore/.local/easybuild/software/gencore_dev/1.0/lib/perl5/site_perl/5.22.0/HPC/Runner/Command/submit_jobs/Utils/Scheduler.pm line 1415.
```

**Comment by jerowe** [comment]
Its a bug for commands_per_node. It is not affecting your submission, but if you want to make it go away before the next release either don't set commands_per_node or set it to 1.

**Comment by jerowe** [comment]
Now that I've investigated it more I realize it will affect your submission. Could you please resubmit any jobs and set commands_per_node=1 ? Or just leave it blank. The default is one. This is a bug and will be fixed in the 3.2.4 release.

**Comment by jerowe** [comment]
This has been fixed in the 3.2.4 release. Closing now.
